blockchain/types/tx_internal_data_serializer.go

```diff
@@ -50,9 +50,17 @@ func (serializer *TxInternalDataSerializer) EncodeRLP(w io.Writer) error {
 		return rlp.Encode(w, serializer.tx)
 	}
 
-	if err := rlp.Encode(w, serializer.txType); err != nil {
-		return err
+	if serializer.txType.IsEthTypedTransaction() {
+		ethType := uint8(serializer.txType)
+		if _, err := w.Write([]byte{TxTypeEthEnvelope, ethType}); err != nil {
```

Comment:
```suggestion
		if _, err := w.Write([]byte{byte(TxTypeEthEnvelope), ethType}); err != nil {
```

If we accept [the previous suggestion](https://github.com/klaytn/klaytn/pull/1164/files#r805378308), we need to convert it to byte type.